src/einsteinpy/bodies.py

```diff
@@ -15,195 +15,110 @@
 Data references can be found in :py:mod:`~einsteinpy.constant`
 """
 
+from dataclasses import dataclass
+from textwrap import dedent
+from typing import Union
+
 from astropy import units as u
 
-from einsteinpy import constant
+from einsteinpy.constant import R_sun, Solar_Mass
+from einsteinpy.coordinates.differential import (
+    BoyerLindquistDifferential,
+    CartesianDifferential,
+    SphericalDifferential,
+)
 
 __all__ = ["Body"]
 
 
+@dataclass
 class Body:
     """

```

Comment:
Change to:
```py
"""
Class to create a generic Body

Parameters
------------
name : str, optional
    Name or ID of the body
    Defaults to ``Generic Body``
mass : ~astropy.units.kg, optional
    Mass of the body
    Defaults to ``0 * u.kg``
q : ~astropy.units.C, optional
    Charge on the body
    Defaults to ``0 * u.C``
R : ~astropy.units.km, optional
    Radius of the body
    Defaults to ``0 * u.km``
coords : ~einsteinpy.coordinates.differential.*, optional
    Complete coordinates of the body
    Defaults to ``None``
parent : Body, optional
    The parent object of the body
    Useful in case of multibody systems
    Defaults to ``None``

"""
```

Reply:
Last change. Will merge after.

Reply:
@JeS24 Updated docstrings